assets/js/theme/common/carousel/index.js

```diff
@@ -5,45 +5,73 @@ import {
     tooltipSetup,
     setTabindexes,
     arrowAriaLabling,
+    updateTextWithLiveData,
     heroCarouselSetup,
     getRealSlidesQuantityAndCurrentSlide,
+    setupAriaAttributes,
 } from './utils';
 
-export const onCarouselChange = (event, carousel) => {
-    const {
-        options: { prevArrow, nextArrow },
-        currentSlide,
-        slideCount,
-        $prevArrow,
-        $nextArrow,
-        $dots,
-        $slider,
-        breakpointSettings,
-        activeBreakpoint,
-    } = carousel;
+/**
+ * returns actualSlide and actualSlideCount
+ * based on provided carousel settings
+ * @param {Object} $slickSettings
+ * @returns {Object}
+ */
+const extractSlidesDetails = ({
+    slideCount, currentSlide, breakpointSettings, activeBreakpoint, $slider,
+}) => getRealSlidesQuantityAndCurrentSlide(
+    breakpointSettings,
+    activeBreakpoint,
+    currentSlide,
+    slideCount,
+    $slider.data('slick').slidesToScroll,
+);
 
-    const { actualSlideCount, actualSlide } = getRealSlidesQuantityAndCurrentSlide(
-        breakpointSettings,
-        activeBreakpoint,
-        currentSlide,
-        slideCount,
-        $slider.data('slick').slidesToScroll,
-    );
+export const onCarouselChange = ({
+    type: eventType, originalEvent, target, data: $activeSlider,
+}, carousel) => {
+    const $parentContainer = $activeSlider.hasClass('productView-thumbnails') ? $('.productView-images') : $activeSlider;
+    switch (eventType) {
+    case 'click': {
+        const $eventTarget = $(target);
+        if (originalEvent.isTrusted && $eventTarget.hasClass('slick-arrow') || $eventTarget.parents().hasClass('slick-dots')) {
+            const { actualSlideCount, actualSlide } = extractSlidesDetails($activeSlider[0].slick);
+            const $carouselContentElement = $('.carousel-content-change-message', $parentContainer);
```

Comment:
as this class used for logic but not for styles, can we add "js-" prefix to it as it is done for example for play/pause button?